crates/plugins/src/prompt.rs

```diff
@@ -0,0 +1,63 @@
+use anyhow::Result;
+use std::io;
+
+/// Prompts user as to whether they trust the source of the plugin and
+/// want to proceed with installation.
+pub(crate) struct Prompter {
+    plugin_name: String,
+    plugin_license: String,
+    source_url: String,
+}
+
+impl Prompter {
+    /// Creates a new prompter
+    pub fn new(plugin_name: &str, plugin_license: &str, source_url: &str) -> Result<Self> {
```

Comment:
As ever, this is subjective, but this design feels a bit odd to me.  A Prompter feels like it would encapsulate a _strategy_ for getting input from the user (e.g. console prompter), and that things like the names would be passed in each time it was called.  With the current design the Prompter object has no useful public role because the only thing you can do is call `run` on it, and `run` returns the result directly - there's no point keeping the Prompter itself around.  Why not have a `prompt(...)` function and avoid the extra characters and concepts of `Prompter:new(...).run()`?

(This isn't me saying "don't do it this way," but me asking for more clarity on _why_ do it this way.)

Reply:
I agree. This can just be a function. changed it!